python/pyiceberg/table/snapshots.py

```diff
@@ -117,12 +125,369 @@ def manifests(self, io: FileIO) -> List[ManifestFile]:
             return list(read_manifest_list(file))
         return []
 
+    def added_data_files(self, io: FileIO) -> Generator[DataFile, None, None]:
+        for manifest in self.manifests(io):
```

Comment:
These aren't necessarily data manifests you're scanning through, so it could include delete files.

Reply:
Ah, great catch!